packages/cozy-sharing/src/helpers/files.js

```diff
@@ -5,9 +5,14 @@ export const fetchFilesPaths = async (client, doctype, files) => {
   const parentDirs = await client
     .collection(doctype)
     .all({ keys: parentDirIds })
-  const filePaths = files.map(f => {
-    const parentDirPath = parentDirs.data.find(d => d.id === f.dir_id).path
-    return parentDirPath === '/' ? `/${f.name}` : `${parentDirPath}/${f.name}`
-  })
+  const filePaths = files
+    .map(f => parentDirs.data.find(d => d.id === f.dir_id))
+    .filter(parentDir => parentDir !== undefined)
+    .map(parentDir => {
+      const parentDirPath = parentDir.path
+      return parentDirPath === '/'
+        ? `/${parentDir.name}`
+        : `${parentDirPath}/${parentDir.name}`
+    })
```

Comment:
This code is not covered by unit tests, so i'm not sure of side effect I may have introduced.

Reply:
let's add a few unit tests so :D